src/psyclone/psyGen.py

```diff
@@ -4199,21 +4184,13 @@ def __init__(self, call, arg_info, access):
             # previous name.
             self._name = self._name_space_manager.create_name(
                 root_name=self._orig_name, context="AlgArgs", label=self._text)
-        # _writers and _readers need to be instances of this class,
-        # rather than static variables, as the mapping that is used
-        # depends on the API and this is only known when a subclass of
-        # Argument is created (as the local MAPPING_ACCESSES will be
-        # used). For example, a dynamo0p3 api instantiation of a
-        # DynArgument (subclass of Argument) will use the
-        # MAPPING_ACCESSES specified in the dynamo0p3 file which
-        # overide the default ones in this file.
-        self._write_access_types = [MAPPING_ACCESSES["write"],
-                                    MAPPING_ACCESSES["readwrite"],
-                                    MAPPING_ACCESSES["inc"],
-                                    MAPPING_REDUCTIONS["sum"]]
-        self._read_access_types = [MAPPING_ACCESSES["read"],
-                                   MAPPING_ACCESSES["readwrite"],
-                                   MAPPING_ACCESSES["inc"]]
+        self._write_access_types = [AccessType.WRITE,
```

Comment:
Could these lists just be defined in AccessType?

Reply:
Sure (there already is 'all_read_access()'). Do you want to keep the self._XXXX_access_types variables, or should I replace them everywhere with corresponding functions in AccessType?

Reply:
I think it would be nice to have everything in one place (`AccessType`) if that works for you.

Reply:
OK, all _{read,write}_access_types are gone now :)